hibernate-core/src/main/java/org/hibernate/graph/FetchAndLoadEntityGraphs.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.graph;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.Supplier;
+
+import javax.persistence.EntityGraph;
+import javax.persistence.EntityManager;
+import javax.persistence.Query;
+
+import org.hibernate.ScrollableResults;
+import org.hibernate.Session;
+import org.hibernate.engine.spi.LoadQueryInfluencers;
+import org.hibernate.engine.spi.SessionImplementor;
+
+/**
+ * Internally used class to carry a pair of graphs (fetch and load), 
+ * a result of {@link EntityGraphParser#parsePreQueryGraphDescriptors(EntityManager, Class, ParseBuffer)}.
+ * 
+ * @author asusnjar
+ *
+ * @param <T> Root entity type of both graphs.
+ */
+class FetchAndLoadEntityGraphs<T> {
```

Comment:
There can only ever be either a fetch graph or a load graph, never both.  So why do you define a class that bundles both?  IMO this really ought to be something more like `org.hibernate.engine.query.spi.EntityGraphQueryHint` with an enum (designating LOAD/FETCH) and the actual graph

Reply:
My understanding was the same but the JP contract statement and interface is not clear about that and does not exclude that case (that I know). Some other developer (ORM or client) could take this, for example, as the result being a union of what the two would independently specify. Thus, we made sure that both can be set. If not valid, the engine underneath would fail it. Additionally, this is just a container that knows what was set and what to pass on. It isn't a public class. If the case is truly invalid, it itself could allow only one to be set.